// tdesign-vue-next/es/hooks/useLazyLoad.mjs
/**
 * tdesign v1.10.5
 * (c) 2024 tdesign
 * @license MIT
 */

import { computed, ref, onMounted, nextTick } from 'vue';
import observe from '../_common/js/utils/observe.mjs';
import { isServer } from '../utils/dom.mjs';
import '../_chunks/dep-eca422c3.mjs';
import '../_chunks/dep-03a7fb6c.mjs';
import '../_chunks/dep-bf7257e7.mjs';
import '../_chunks/dep-4659d73b.mjs';
import '../_chunks/dep-eae2a67e.mjs';
import '../_chunks/dep-626c497d.mjs';
import '../_chunks/dep-29ef8419.mjs';
import '../_chunks/dep-06dcbf4c.mjs';
import '../_chunks/dep-2b09875b.mjs';
import '../_chunks/dep-cda4f568.mjs';
import '../utils/easing.mjs';

function useLazyLoad(containerRef, childRef, params) {
  var tRowHeight = computed(function () {
    return Math.max(params.rowHeight || 48, 48);
  });
  var isInit = ref(false);
  var hasLazyLoadHolder = computed(function () {
    return (params === null || params === void 0 ? void 0 : params.type) === "lazy" && !isInit.value;
  });
  var requestAnimationFrame = !isServer && window.requestAnimationFrame || function (cb) {
    return setTimeout(cb, 16.6);
  };
  var init = function init() {
    if (!isInit.value) {
      requestAnimationFrame(function () {
        isInit.value = true;
      });
    }
  };
  onMounted(function () {
    if ((params === null || params === void 0 ? void 0 : params.type) !== "lazy") return;
    nextTick(function () {
      var bufferSize = Math.max(10, params.bufferSize || 10);
      var height = tRowHeight.value * bufferSize;
      observe(childRef.value, containerRef.value, init, height);
    });
  });
  return {
    hasLazyLoadHolder: hasLazyLoadHolder,
    tRowHeight: tRowHeight
  };
}

export { useLazyLoad as default };
