// tdesign-vue-next/cjs/menu/menu-group.js
/**
 * tdesign v1.10.5
 * (c) 2024 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var menu_menuGroupProps = require('./menu-group-props.js');
var utils_renderTnode = require('../utils/render-tnode.js');
var hooks_useConfig = require('../hooks/useConfig.js');
require('lodash/isEmpty');
require('lodash/isString');
require('lodash/isFunction');
require('lodash/isObject');
require('lodash/camelCase');
require('lodash/kebabCase');
require('lodash/isArray');
require('../config-provider/useConfig.js');
require('lodash/cloneDeep');
require('../config-provider/context.js');
require('lodash/mergeWith');
require('lodash/merge');
require('../_common/js/global-config/default-config.js');
require('../_common/js/global-config/locale/zh_CN.js');
require('../_chunks/dep-14f69041.js');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-0a56babd.js');
require('dayjs');

var _MenuGroup = Vue.defineComponent({
  name: "TMenuGroup",
  props: menu_menuGroupProps["default"],
  setup: function setup() {
    var classPrefix = hooks_useConfig.usePrefixClass();
    return {
      classPrefix: classPrefix
    };
  },
  render: function render() {
    var classPrefix = this.classPrefix;
    return Vue.createVNode("div", {
      "class": "".concat(classPrefix, "-menu-group")
    }, [Vue.createVNode("div", {
      "class": "".concat(classPrefix, "-menu-group__title")
    }, [utils_renderTnode.renderTNodeJSX(this, "title", {
      silent: false
    })]), utils_renderTnode.renderTNodeJSX(this, "default")]);
  }
});

exports["default"] = _MenuGroup;
