// tdesign-vue-next/esm/input/props.js
/**
 * tdesign v1.10.5
 * (c) 2024 tdesign
 * @license MIT
 */

var props = {
  align: {
    type: String,
    "default": "left",
    validator: function validator(val) {
      if (!val) return true;
      return ["left", "center", "right"].includes(val);
    }
  },
  allowInputOverMax: Boolean,
  autoWidth: Boolean,
  autocomplete: {
    type: String,
    "default": void 0
  },
  autofocus: Boolean,
  borderless: Boolean,
  clearable: Boolean,
  disabled: {
    type: Boolean,
    "default": void 0
  },
  format: {
    type: Function
  },
  inputClass: {
    type: [String, Object, Array]
  },
  label: {
    type: [String, Function]
  },
  maxcharacter: {
    type: Number
  },
  maxlength: {
    type: [String, Number]
  },
  name: {
    type: String,
    "default": ""
  },
  placeholder: {
    type: String,
    "default": void 0
  },
  prefixIcon: {
    type: Function
  },
  readonly: {
    type: Boolean,
    "default": void 0
  },
  showClearIconOnEmpty: Boolean,
  showLimitNumber: Boolean,
  size: {
    type: String,
    "default": "medium",
    validator: function validator(val) {
      if (!val) return true;
      return ["small", "medium", "large"].includes(val);
    }
  },
  spellCheck: Boolean,
  status: {
    type: String,
    "default": "default",
    validator: function validator(val) {
      if (!val) return true;
      return ["default", "success", "warning", "error"].includes(val);
    }
  },
  suffix: {
    type: [String, Function]
  },
  suffixIcon: {
    type: Function
  },
  tips: {
    type: [String, Function]
  },
  type: {
    type: String,
    "default": "text",
    validator: function validator(val) {
      if (!val) return true;
      return ["text", "number", "url", "tel", "password", "search", "submit", "hidden"].includes(val);
    }
  },
  value: {
    type: [String, Number],
    "default": void 0
  },
  modelValue: {
    type: [String, Number],
    "default": void 0
  },
  defaultValue: {
    type: [String, Number]
  },
  onBlur: Function,
  onChange: Function,
  onClear: Function,
  onClick: Function,
  onCompositionend: Function,
  onCompositionstart: Function,
  onEnter: Function,
  onFocus: Function,
  onKeydown: Function,
  onKeypress: Function,
  onKeyup: Function,
  onMouseenter: Function,
  onMouseleave: Function,
  onPaste: Function,
  onValidate: Function,
  onWheel: Function
};

export { props as default };
