// tdesign-vue-next/es/steps/props.mjs
/**
 * tdesign v1.10.5
 * (c) 2024 tdesign
 * @license MIT
 */

var props = {
  current: {
    type: [String, Number],
    "default": void 0
  },
  modelValue: {
    type: [String, Number],
    "default": void 0
  },
  defaultCurrent: {
    type: [String, Number]
  },
  layout: {
    type: String,
    "default": "horizontal",
    validator: function validator(val) {
      if (!val) return true;
      return ["horizontal", "vertical"].includes(val);
    }
  },
  options: {
    type: Array
  },
  readonly: {
    type: Boolean,
    "default": void 0
  },
  separator: {
    type: String,
    "default": "line",
    validator: function validator(val) {
      if (!val) return true;
      return ["line", "dashed", "arrow"].includes(val);
    }
  },
  sequence: {
    type: String,
    "default": "positive",
    validator: function validator(val) {
      if (!val) return true;
      return ["positive", "reverse"].includes(val);
    }
  },
  theme: {
    type: String,
    "default": "default",
    validator: function validator(val) {
      if (!val) return true;
      return ["default", "dot"].includes(val);
    }
  },
  onChange: Function
};

export { props as default };
