// tdesign-vue-next/esm/config-provider/context.js
/**
 * tdesign v1.10.5
 * (c) 2024 tdesign
 * @license MIT
 */

import _mergeWith from 'lodash/mergeWith';
import merge from 'lodash/merge';
import defaultConfig from '../_common/js/global-config/default-config.js';
import zhCn from '../_common/js/global-config/locale/zh_CN.js';
import isArray from 'lodash/isArray';
import '../_chunks/dep-27a6bfe1.js';
import '@babel/runtime/helpers/typeof';
import '../_chunks/dep-cf1a698d.js';
import 'dayjs';
import '../_chunks/dep-4ad70462.js';

var EAnimationType = /* @__PURE__ */function (EAnimationType2) {
  EAnimationType2["ripple"] = "ripple";
  EAnimationType2["expand"] = "expand";
  EAnimationType2["fade"] = "fade";
  return EAnimationType2;
}(EAnimationType || {});
var defaultGlobalConfig = merge(defaultConfig, zhCn);
var configProviderInjectKey = Symbol("configProvide");
var mergeWith = function mergeWith(defaultGlobalConfig2, injectConfig) {
  return _mergeWith(defaultGlobalConfig2, injectConfig, function (objValue, srcValue) {
    if (isArray(objValue)) {
      return srcValue;
    }
  });
};

export { EAnimationType, configProviderInjectKey, defaultGlobalConfig, mergeWith };
