// tdesign-vue-next/es/input/useInputEventHandler.mjs
/**
 * tdesign v1.10.5
 * (c) 2024 tdesign
 * @license MIT
 */

import { getOutputValue } from './useInput.mjs';
import '../_chunks/dep-5efe53d1.mjs';
import '../_chunks/dep-593f2b67.mjs';
import 'vue';
import '../form/const.mjs';
import '../hooks/index.mjs';
import '../hooks/icon.mjs';
import '../_chunks/dep-06dcbf4c.mjs';
import '../_chunks/dep-eca422c3.mjs';
import '../_chunks/dep-bf7257e7.mjs';
import '../_chunks/dep-4659d73b.mjs';
import '../_chunks/dep-2b09875b.mjs';
import '../_chunks/dep-cda4f568.mjs';
import '../hooks/slot.mjs';
import '../_chunks/dep-eae2a67e.mjs';
import '../utils/render-tnode.mjs';
import '../_chunks/dep-b91a8171.mjs';
import '../_chunks/dep-1b78414a.mjs';
import '../_chunks/dep-1a4bb2f2.mjs';
import '../_chunks/dep-cc8ebcde.mjs';
import '../_chunks/dep-76958824.mjs';
import '../_chunks/dep-29ef8419.mjs';
import '../_chunks/dep-f5579c03.mjs';
import '../_chunks/dep-6dc3d0e1.mjs';
import '../_chunks/dep-d84f19b3.mjs';
import '../_chunks/dep-3c65fae8.mjs';
import '../_chunks/dep-626c497d.mjs';
import '../_chunks/dep-03a7fb6c.mjs';
import '../_chunks/dep-03ad9cda.mjs';
import '../_chunks/dep-7932c2e8.mjs';
import '../_chunks/dep-ab4bb9b8.mjs';
import '../_chunks/dep-d8a34055.mjs';
import '../_chunks/dep-2f6e28a0.mjs';
import '../_chunks/dep-5a17bf21.mjs';
import '../_chunks/dep-bad871d5.mjs';
import '../hooks/tnode.mjs';
import '../hooks/useConfig.mjs';
import '../config-provider/useConfig.mjs';
import '../_chunks/dep-bb899bfd.mjs';
import '../_chunks/dep-5922c8f3.mjs';
import '../_chunks/dep-92f86d1d.mjs';
import '../_chunks/dep-6ab5363f.mjs';
import '../_chunks/dep-e9841b9e.mjs';
import '../_chunks/dep-49aeee3c.mjs';
import '../_chunks/dep-b5bf85f5.mjs';
import '../_chunks/dep-ec0bbd9f.mjs';
import '../_chunks/dep-732f0b7d.mjs';
import '../_chunks/dep-1b9718a2.mjs';
import '../_chunks/dep-e065dc25.mjs';
import '../_chunks/dep-400ec705.mjs';
import '../_chunks/dep-02f9394e.mjs';
import '../_chunks/dep-1725b1c7.mjs';
import '../_chunks/dep-532409c5.mjs';
import '../_common/js/global-config/default-config.mjs';
import '../_common/js/global-config/locale/zh_CN.mjs';
import '../_chunks/dep-9ed7da3e.mjs';
import '../_chunks/dep-7f65fc46.mjs';
import '../_chunks/dep-4f2cabb0.mjs';
import '../config-provider/type.mjs';
import '../hooks/useCommonClassName.mjs';
import '../hooks/useDefaultValue.mjs';
import '../hooks/useDestroyOnClose.mjs';
import '../hooks/useKeepAnimation.mjs';
import '../hooks/useRipple.mjs';
import '../utils/set-style.mjs';
import '../hooks/useVirtualScroll.mjs';
import '../_chunks/dep-fa52aa21.mjs';
import '../hooks/useVModel.mjs';
import '../hooks/useImagePreviewUrl.mjs';
import '../_common/js/upload/utils.mjs';
import '../_common/js/log/log.mjs';
import '../hooks/useDisabled.mjs';
import '../_chunks/dep-df472119.mjs';
import './useLengthLimit.mjs';
import '../_common/js/log/index.mjs';
import '../_common/js/utils/helper.mjs';
import '../_chunks/dep-4a990ea3.mjs';
import '../_chunks/dep-5d3f7c0e.mjs';
import '../_chunks/dep-73384992.mjs';
import '../_chunks/dep-58a62202.mjs';
import '../_chunks/dep-889e457b.mjs';
import '../hooks/useReadonly.mjs';

function useInputEventHandler(props, isHover) {
  var handleKeydown = function handleKeydown(e) {
    if (props.disabled) return;
    var code = e.code;
    var tmpValue = getOutputValue(e.currentTarget.value, props.type);
    if (/enter/i.test(code) || /enter/i.test(e.key)) {
      var _props$onEnter;
      (_props$onEnter = props.onEnter) === null || _props$onEnter === void 0 || _props$onEnter.call(props, tmpValue, {
        e: e
      });
    } else {
      var _props$onKeydown;
      (_props$onKeydown = props.onKeydown) === null || _props$onKeydown === void 0 || _props$onKeydown.call(props, tmpValue, {
        e: e
      });
    }
  };
  var handleKeyUp = function handleKeyUp(e) {
    var _props$onKeyup;
    if (props.disabled) return;
    var tmpValue = getOutputValue(e.currentTarget.value, props.type);
    (_props$onKeyup = props.onKeyup) === null || _props$onKeyup === void 0 || _props$onKeyup.call(props, tmpValue, {
      e: e
    });
  };
  var handleKeypress = function handleKeypress(e) {
    var _props$onKeypress;
    if (props.disabled) return;
    var tmpValue = getOutputValue(e.currentTarget.value, props.type);
    (_props$onKeypress = props.onKeypress) === null || _props$onKeypress === void 0 || _props$onKeypress.call(props, tmpValue, {
      e: e
    });
  };
  var onHandlePaste = function onHandlePaste(e) {
    var _props$onPaste;
    if (props.disabled) return;
    var clipData = e.clipboardData || window.clipboardData;
    (_props$onPaste = props.onPaste) === null || _props$onPaste === void 0 || _props$onPaste.call(props, {
      e: e,
      pasteValue: clipData === null || clipData === void 0 ? void 0 : clipData.getData("text/plain")
    });
  };
  var mouseEvent = function mouseEvent(v) {
    return isHover.value = v;
  };
  var onHandleMousewheel = function onHandleMousewheel(e) {
    var _props$onWheel;
    return (_props$onWheel = props.onWheel) === null || _props$onWheel === void 0 ? void 0 : _props$onWheel.call(props, {
      e: e
    });
  };
  var onInputMouseenter = function onInputMouseenter(e) {
    var _props$onMouseenter;
    mouseEvent(true);
    (_props$onMouseenter = props.onMouseenter) === null || _props$onMouseenter === void 0 || _props$onMouseenter.call(props, {
      e: e
    });
  };
  var onInputMouseleave = function onInputMouseleave(e) {
    var _props$onMouseleave;
    mouseEvent(false);
    (_props$onMouseleave = props.onMouseleave) === null || _props$onMouseleave === void 0 || _props$onMouseleave.call(props, {
      e: e
    });
  };
  return {
    isHover: isHover,
    handleKeydown: handleKeydown,
    handleKeyUp: handleKeyUp,
    handleKeypress: handleKeypress,
    onHandlePaste: onHandlePaste,
    onHandleMousewheel: onHandleMousewheel,
    onInputMouseenter: onInputMouseenter,
    onInputMouseleave: onInputMouseleave
  };
}

export { useInputEventHandler as default };
