// tdesign-vue-next/es/_common/js/tree-v1/tree-node-model.mjs
/**
 * tdesign v1.10.5
 * (c) 2024 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../../../_chunks/dep-fa52aa21.mjs';
import { i as isUndefined_1 } from '../../../_chunks/dep-58a62202.mjs';
import { i as isBoolean_1 } from '../../../_chunks/dep-df472119.mjs';
import { p as pick_1 } from '../../../_chunks/dep-4de7e2ca.mjs';
import { o as omit_1 } from '../../../_chunks/dep-88ae49da.mjs';
import log from '../log/log.mjs';
import '../../../_chunks/dep-4659d73b.mjs';
import '../../../_chunks/dep-eca422c3.mjs';
import '../../../_chunks/dep-bf7257e7.mjs';
import '../../../_chunks/dep-29ef8419.mjs';
import '../../../_chunks/dep-aeaef131.mjs';
import '../../../_chunks/dep-eae2a67e.mjs';
import '../../../_chunks/dep-626c497d.mjs';
import '../../../_chunks/dep-ab4bb9b8.mjs';
import '../../../_chunks/dep-6ab5363f.mjs';
import '../../../_chunks/dep-6dc3d0e1.mjs';
import '../../../_chunks/dep-06dcbf4c.mjs';
import '../../../_chunks/dep-2b09875b.mjs';
import '../../../_chunks/dep-cda4f568.mjs';
import '../../../_chunks/dep-76958824.mjs';
import '../../../_chunks/dep-e9841b9e.mjs';
import '../../../_chunks/dep-7932c2e8.mjs';
import '../../../_chunks/dep-0e581ac0.mjs';
import '../../../_chunks/dep-49aeee3c.mjs';
import '../../../_chunks/dep-b5bf85f5.mjs';
import '../../../_chunks/dep-ec0bbd9f.mjs';
import '../../../_chunks/dep-69e0f6fc.mjs';
import '../../../_chunks/dep-3c65fae8.mjs';
import '../../../_chunks/dep-cc8ebcde.mjs';
import '../../../_chunks/dep-4ec29b23.mjs';
import '../../../_chunks/dep-732f0b7d.mjs';
import '../../../_chunks/dep-02f9394e.mjs';
import '../../../_chunks/dep-400ec705.mjs';
import '../../../_chunks/dep-5922c8f3.mjs';
import '../../../_chunks/dep-92f86d1d.mjs';
import '../../../_chunks/dep-1a4bb2f2.mjs';
import '../../../_chunks/dep-f5579c03.mjs';
import '../../../_chunks/dep-1b78414a.mjs';
import '../../../_chunks/dep-d84f19b3.mjs';
import '../../../_chunks/dep-88c8d9ed.mjs';
import '../../../_chunks/dep-2f6e28a0.mjs';
import '../../../_chunks/dep-1725b1c7.mjs';

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function getExposedProps(node) {
  var props = pick_1(node, ["value", "label", "data", "actived", "expanded", "checked", "indeterminate", "loading"]);
  return props;
}
function createNodeModel(node) {
  var props = getExposedProps(node);
  var model = _objectSpread(_objectSpread({}, props), {}, {
    getLevel: function getLevel() {
      return node.getLevel();
    },
    getIndex: function getIndex() {
      return node.getIndex();
    },
    isFirst: function isFirst() {
      return node.isFirst();
    },
    isLast: function isLast() {
      return node.isLast();
    },
    isLeaf: function isLeaf() {
      return node.isLeaf();
    },
    insertBefore: function insertBefore(newData) {
      return node.insertBefore(newData);
    },
    insertAfter: function insertAfter(newData) {
      return node.insertAfter(newData);
    },
    appendData: function appendData(data) {
      return node.append(data);
    },
    getPath: function getPath() {
      var nodes = node.getPath();
      return nodes.map(function (item) {
        return item.getModel();
      });
    },
    getParent: function getParent() {
      var _node$parent;
      return (_node$parent = node.parent) === null || _node$parent === void 0 ? void 0 : _node$parent.getModel();
    },
    getParents: function getParents() {
      var nodes = node.getParents();
      return nodes.map(function (item) {
        return item.getModel();
      });
    },
    getRoot: function getRoot() {
      var root = node.getRoot();
      return root === null || root === void 0 ? void 0 : root.getModel();
    },
    getSiblings: function getSiblings() {
      var nodes = node.getSiblings();
      return nodes.map(function (item) {
        return item.getModel();
      });
    },
    getChildren: function getChildren(deep) {
      var childrenModel = false;
      var children = node.children;
      if (Array.isArray(children)) {
        if (children.length > 0) {
          if (deep) {
            var nodes = node.walk();
            nodes.shift();
            childrenModel = nodes.map(function (item) {
              return item.getModel();
            });
          } else {
            childrenModel = children.map(function (item) {
              return item.getModel();
            });
          }
        } else {
          childrenModel = false;
        }
      } else if (isBoolean_1(children)) {
        childrenModel = children;
      }
      return childrenModel;
    },
    remove: function remove(value) {
      if (!value) {
        node.remove();
        return;
      }
      var tree = node.tree;
      var targetNode = tree.getNode(value);
      if (!targetNode) {
        log.warnOnce("Tree", "`".concat(value, "` is not exist"));
        return;
      }
      var parents = targetNode.getParents();
      var parentValues = parents.map(function (pnode) {
        return pnode.value;
      });
      if (parentValues.indexOf(node.value) < 0) {
        log.warnOnce("Tree", "`".concat(value, "` is not a childNode of current node"));
        return;
      }
      targetNode.remove();
    },
    setData: function setData(data) {
      var _data = omit_1(data, ["children", "value", "label"]);
      var keys = node.tree.config.keys;
      var dataValue = data[(keys === null || keys === void 0 ? void 0 : keys.value) || "value"];
      var dataLabel = data[(keys === null || keys === void 0 ? void 0 : keys.label) || "label"];
      if (!isUndefined_1(dataValue)) _data.value = dataValue;
      if (!isUndefined_1(dataLabel)) _data.label = dataLabel;
      Object.assign(node.data, _data);
      Object.assign(node, _data);
    }
  });
  return model;
}
function updateNodeModel(model, node) {
  var props = getExposedProps(node);
  Object.assign(model, props);
}

export { createNodeModel, updateNodeModel };
