// tdesign-vue-next/es/config-provider/useConfig.mjs
/**
 * tdesign v1.10.5
 * (c) 2024 tdesign
 * @license MIT
 */

import { ref, getCurrentInstance, inject, computed, h, provide } from 'vue';
import { i as isFunction_1 } from '../_chunks/dep-06dcbf4c.mjs';
import { c as cloneDeep_1 } from '../_chunks/dep-bb899bfd.mjs';
import { i as isString_1 } from '../_chunks/dep-03a7fb6c.mjs';
import { c as configProviderInjectKey, d as defaultGlobalConfig, a as mergeWith } from '../_chunks/dep-1b9718a2.mjs';
import './type.mjs';
import '../_chunks/dep-eca422c3.mjs';
import '../_chunks/dep-bf7257e7.mjs';
import '../_chunks/dep-4659d73b.mjs';
import '../_chunks/dep-2b09875b.mjs';
import '../_chunks/dep-cda4f568.mjs';
import '../_chunks/dep-5922c8f3.mjs';
import '../_chunks/dep-92f86d1d.mjs';
import '../_chunks/dep-6ab5363f.mjs';
import '../_chunks/dep-6dc3d0e1.mjs';
import '../_chunks/dep-76958824.mjs';
import '../_chunks/dep-e9841b9e.mjs';
import '../_chunks/dep-1a4bb2f2.mjs';
import '../_chunks/dep-cc8ebcde.mjs';
import '../_chunks/dep-29ef8419.mjs';
import '../_chunks/dep-f5579c03.mjs';
import '../_chunks/dep-49aeee3c.mjs';
import '../_chunks/dep-b5bf85f5.mjs';
import '../_chunks/dep-3c65fae8.mjs';
import '../_chunks/dep-eae2a67e.mjs';
import '../_chunks/dep-626c497d.mjs';
import '../_chunks/dep-ec0bbd9f.mjs';
import '../_chunks/dep-1b78414a.mjs';
import '../_chunks/dep-d84f19b3.mjs';
import '../_chunks/dep-732f0b7d.mjs';
import '../_chunks/dep-e065dc25.mjs';
import '../_chunks/dep-400ec705.mjs';
import '../_chunks/dep-02f9394e.mjs';
import '../_chunks/dep-1725b1c7.mjs';
import '../_chunks/dep-532409c5.mjs';
import '../_common/js/global-config/default-config.mjs';
import '../_common/js/global-config/locale/zh_CN.mjs';
import '../_chunks/dep-9ed7da3e.mjs';
import '../_chunks/dep-7f65fc46.mjs';
import '../_chunks/dep-4f2cabb0.mjs';

var globalConfigCopy = ref();
function useConfig() {
  var componentName = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : void 0;
  var componentLocale = arguments.length > 1 ? arguments[1] : undefined;
  var injectGlobalConfig = getCurrentInstance() ? inject(configProviderInjectKey, null) : globalConfigCopy;
  var mergedGlobalConfig = computed(function () {
    return (injectGlobalConfig === null || injectGlobalConfig === void 0 ? void 0 : injectGlobalConfig.value) || defaultGlobalConfig;
  });
  var globalConfig = computed(function () {
    return Object.assign({}, mergedGlobalConfig.value[componentName], componentLocale);
  });
  var classPrefix = computed(function () {
    return mergedGlobalConfig.value.classPrefix;
  });
  var t = function t(pattern) {
    for (var _len = arguments.length, args = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
      args[_key - 1] = arguments[_key];
    }
    var data = args[0];
    if (isString_1(pattern)) {
      if (!data) return pattern;
      var regular = /\{\s*([\w-]+)\s*\}/g;
      var translated = pattern.replace(regular, function (match, key) {
        if (data) {
          return String(data[key]);
        }
        return "";
      });
      return translated;
    }
    if (isFunction_1(pattern)) {
      if (!args.length) return pattern(h);
      return pattern.apply(void 0, args);
    }
    return "";
  };
  return {
    t: t,
    global: globalConfig,
    globalConfig: globalConfig,
    classPrefix: classPrefix
  };
}
var provideConfig = function provideConfig(props) {
  var defaultData = cloneDeep_1(defaultGlobalConfig);
  var mergedGlobalConfig = computed(function () {
    return Object.assign({}, mergeWith(defaultData, props.globalConfig));
  });
  provide(configProviderInjectKey, mergedGlobalConfig);
  if (!globalConfigCopy.value) {
    globalConfigCopy.value = mergedGlobalConfig.value;
  }
  return mergedGlobalConfig;
};

export { provideConfig, useConfig };
