// tdesign-vue-next/cjs/_common/js/date-picker/utils.js
/**
 * tdesign v1.10.5
 * (c) 2024 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('@babel/runtime/helpers/slicedToArray');
require('@babel/runtime/helpers/objectDestructuringEmpty');
require('@babel/runtime/helpers/extends');
require('@babel/runtime/helpers/defineProperty');
require('lodash/isFunction');
require('dayjs');
var _common_js_datePicker_utils = require('../../../_chunks/dep-06f97ca1.js');
require('../../../_chunks/dep-e2f5caca.js');
require('lodash/chunk');
require('lodash/isString');
require('../../../_chunks/dep-0a56babd.js');
require('../log/log.js');



exports.addMonth = _common_js_datePicker_utils.addMonth;
exports.covertToDate = _common_js_datePicker_utils.covertToDate;
exports.extractTimeFormat = _common_js_datePicker_utils.extractTimeFormat;
exports.extractTimeObj = _common_js_datePicker_utils.extractTimeObj;
exports.firstUpperCase = _common_js_datePicker_utils.firstUpperCase;
exports.flagActive = _common_js_datePicker_utils.flagActive;
exports.getDateObj = _common_js_datePicker_utils.getDateObj;
exports.getMonths = _common_js_datePicker_utils.getMonths;
exports.getQuarters = _common_js_datePicker_utils.getQuarters;
exports.getToday = _common_js_datePicker_utils.getToday;
exports.getWeeks = _common_js_datePicker_utils.getWeeks;
exports.getYears = _common_js_datePicker_utils.getYears;
exports.isEnabledDate = _common_js_datePicker_utils.isEnabledDate;
exports.isSame = _common_js_datePicker_utils.isSame;
exports.outOfRanges = _common_js_datePicker_utils.outOfRanges;
exports.setDateTime = _common_js_datePicker_utils.setDateTime;
exports.subtractMonth = _common_js_datePicker_utils.subtractMonth;
